### ISSUE TITLE
feature: add ability to have in one config file multiple configs for different paths

### ISSUE THREAD

**Issue by marcindabrowski** [adrienverge/yamllint#670]
It would be great to give ability to define different configuration for different paths in one config file, ie:

```yaml
---

extends: default

yaml-files:
  - '*.yaml'
  - '*.yml'
  - '.yamllint'

rules:
  # default rules for all files in directory

---

yaml-files:
  - 'folder1/*.yaml'

rules:
  # rules that overrides some config flags for all yamls in folder1
```

I'm aware that I can run `yamllint` with different configs for different sets of files, but this approach could simplify the usage.

Why I want this change?
Because the problem with `on` key in GitHub Workflows. I would like to have ability to change one config rule for given path.

**Comment by adrienverge** [comment]
Hello, and thanks for the clear proposal :+1:

I'm not sure the need is worth creating a feature to parse different YAML documents from a single config file, and handle their potential conflicts.

> Why I want this change?
> Because the problem with `on` key in GitHub Workflows. I would like to have ability to change one config rule for given path.

Have you considered one of these [4 solutions to allow `on` key in GitHub workflows](https://github.com/adrienverge/yamllint/issues/666#issuecomment-2034469482)?

**Comment by gecube** [comment]
+1 on this. We hit this exact problem in a GitOps repo (Kubernetes/Flux) with ~1,200 YAML files. K8s manifests need `document-start` enforced, but `values.yaml`, `kustomization.yaml`, and SOPS-encrypted `secret-values.yaml` don't — and today the only workaround is maintaining two config files + a CI script that greps filenames to route them to the right one.

I wrote up a detailed proposal with concrete syntax, semantics, and prior art references here: https://github.com/owenlamont/ryl/issues/191

The Rust-based alternatives are gaining traction but none implement this either — yamllint could stay ahead by adding it natively. Happy to contribute a PR if there's interest.

**Comment by miettal** [comment]
I'm facing the same issue with VCR cassette files.
I implemented the overrides proposal from https://github.com/owenlamont/ryl/issues/191. How about this?
https://github.com/adrienverge/yamllint/compare/master...miettal:yamllint:feat/path-scoped-overrides?expand=1
